### PR TITLE
new patched SPM12 version 

### PR DESCRIPTION
**Pull request by TheChymera**
As discussed in #499

### CODE REVIEW

**Comment by jlec** [comment]
from http://www.fil.ion.ucl.ac.uk/spm/download/spm12_updates/ it looks like patches are from rev 6470. How about naming the version _p6470?

**Comment by TheChymera** [comment]
ok, why not `-r6470` ?

**Comment by marbre** [comment]
You should really read https://devmanual.gentoo.org/ebuild-writing/file-format/index.html.

**Comment by marbre** [comment]
Further https://devmanual.gentoo.org/general-concepts/ebuild-revisions/index.html tells:
`Ebuilds may have a Gentoo revision number associated with them. This is a -rX suffix, where X is an integer — see File Naming Rules. This component must only be used for Gentoo changes, not upstream releases. By default, -r0 is implied. `

It is all in the docs...

**Comment by TheChymera** [comment]
got it.

File: sci-biology/spm/spm-12_p6470.ebuild [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+EAPI=5
+
+inherit versionator
+
+MY_PV=$(get_major_version)
+
+DESCRIPTION="Analysis of brain imaging data sequences for Octave or Matlab"
+HOMEPAGE="http://www.fil.ion.ucl.ac.uk/spm/"
+SRC_URI="http://www.fil.ion.ucl.ac.uk/spm/download/restricted/eldorado/${PN}${MY_PV}.zip"
```
Comment (jlec) [inline sci-biology/spm/spm-12_p6470.ebuild]: You should do the following here
```
SRC_URI="http://www.fil.ion.ucl.ac.uk/spm/download/restricted/eldorado/${PN}${MY_PV}.zip -> ${P}.zip
```

**Comment by TheChymera** [comment]
@jlec Looking good now?

File: sci-biology/spm/spm-12_p6470.ebuild [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+EAPI=5
+
+inherit versionator
+
+MY_PV=$(get_major_version)
+
+DESCRIPTION="Analysis of brain imaging data sequences for Octave or Matlab"
+HOMEPAGE="http://www.fil.ion.ucl.ac.uk/spm/"
+SRC_URI="http://www.fil.ion.ucl.ac.uk/spm/download/restricted/eldorado/${PN}${MY_PV}.zip -> ${P}"
```
Comment (marbre) [inline sci-biology/spm/spm-12_p6470.ebuild]: filetype missing, you need `-> ${P}.zip`

Comment (marbre) [inline sci-biology/spm/spm-12_p6470.ebuild]: Further, your downloading `spm12.zip`, but the update `spm12_updates_r6470.zip` from http://www.fil.ion.ucl.ac.uk/spm/download/spm12_updates/. isn't fetched. You should check, whether `spm12.zip` + the update is still needed or if the full source is provided with `spm12_updates_r6470.zip`.

**Comment by TheChymera** [comment]
`spm-12.zip` atm is `spm-12.0.zip` plus the updates from `spm12_updates_r6470.zip`.

File: sci-biology/spm/spm-12_p6470.ebuild (hunk 2 above)
Comment (TheChymera) [inline sci-biology/spm/spm-12_p6470.ebuild]: `spm-12.zip` atm is `spm-12.0.zip` plus the updates from `spm12_updates_r6470.zip`.

**Pull request merged**